src/sgame/sg_active.cpp

```diff
@@ -2184,10 +2184,11 @@ void ClientThink_real( gentity_t *self )
 
 		ent = &g_entities[ trace.entityNum ];
 
-		if ( ent && ent->use &&
-		     ( !ent->buildableTeam   || ent->buildableTeam   == client->pers.team ) &&
-		     ( !ent->conditions.team || ent->conditions.team == client->pers.team ) &&
-		     Distance( self->s.origin, ent->s.origin ) < ENTITY_USE_RANGE )
+		if ( ent && ent->use
+		     && ( !ent->buildableTeam   || ent->buildableTeam   == client->pers.team )
+		     && ( !ent->conditions.team || ent->conditions.team == client->pers.team )
+		     && trace.fraction < 1.0f
```

Comment:
How about keeping the distance check only for buildables? Something like that (but redacted to be more readable and use the correct symbol names — I am typing this on a phone, can't grep)

```suggestion
		     && !( entityType == eType_t::BUILDABLE && Distance( self->s.origin, ent->s.origin ) >= ENTITY_USE_RANGE )
```

So that this would fix the use of map entities without making the buildable case inconsistent anew

Reply:
I think this is a reasonable compromise. I think it's fair to have different modes of "usable" entities. Not all usable entities need to behave the same, especially if the usable entities we are trying to support are legacy entities.